### PR TITLE
feat: setup explorer on argus

### PR DESCRIPTION
**Pull request by jakeyheath**
## Summary

This is a demo PR of what the workflow and new configuration for the argus platform would look like. All new infra configuration is in the `.infra` folder and split by environment (very similar to how `.happy` looks). However, instead of a terraform files, there are YAML configuration files. These files look more like native kubernetes setting and load a Helm chart. 

## Details

* `.infra/common.yaml` - Configuration for all environments. Think of this as your base set of common configuration.
* `.infra/rdev/values.yaml` - Configuration overridees for the `rdev` environment. These settings can add or overwrite the configurations set in `.infra/common.yaml`.

## References

* The helm chart source is [here](https://github.com/chanzuckerberg/argo-helm-charts/blob/main/stack/values.yaml).

## Notes

* There is an open bug in tiledb that isn't allowing us to use the pod's EKS IAM role. For now, we have manually made an AWS user and associated it with the policy and stored the credentials in the K8S secrets.

### CODE REVIEW

File: .github/workflows/docker-build.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+    workflow_dispatch: {}
+    push:
+        paths-ignore:
+            - '.github/**'
+            - '.happy/**'
+            - 'dev_docs/**'
+            - 'example-dataset/**'
+        branches-ignore:
+          - main
```
Comment (jakeyheath) [inline .github/workflows/docker-build.yaml:12]: - only rebuild if the source changes
- don't build and push on the main branch (the container image should already be built by the time it merges to main)

File: hosted/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:22.04
+FROM 471112759938.dkr.ecr.us-west-2.amazonaws.com/docker.io/sci/library/ubuntu:22.04
```
Comment (jakeyheath) [inline hosted/Dockerfile:1]: Utilizes our ECR pull through cache so we don't accidentally hit dockerhub rate limit issus. Also allows us to do security scanning and other security monitoring on the images we pull.

File: .github/workflows/docker-build.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            - '.infra/**'
+            - '.github/**'
+            - '.happy/**'
+            - 'dev_docs/**'
+            - 'example-dataset/**'
+        branches-ignore:
+          - main
+
+jobs:
+  argus_builder:
```
Comment (jakeyheath) [inline .github/workflows/docker-build.yaml:20]: Docker builds are done through this CI action. Once the image is built, the image tag is committed back to the PR. This tag will be used to deploy the stack.

File: .infra/rdev/values.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+services:
+  explorer:
+    image:
+      tag: sha-652d1618
```
Comment (jakeyheath) [inline .infra/rdev/values.yaml:4]: this is the current container tag used for deployement. Whatever is checked into the repo is what is deployed

File: .github/workflows/argus-create-stack-rdev.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  create-rdev-stack:
+    runs-on: [ARM64, self-hosted, Linux]
+    if: contains(github.event.pull_request.labels.*.name, 'stack')
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Create Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.16.0
```
Comment (hspitzley-czi) [inline .github/workflows/argus-create-stack-rdev.yaml:17]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.17.1
```

File: .github/workflows/argus-delete-stack-rdev.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  delete-rdev-stack:
+    runs-on: [ARM64, self-hosted, Linux]
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'stack') || github.event.pull_request.state == 'closed' }}
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Delete Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/delete-stack@v0.16.0
```
Comment (hspitzley-czi) [inline .github/workflows/argus-delete-stack-rdev.yaml:17]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/delete-stack@v0.17.1
```

File: .infra/common.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      - name: CXG_BUCKET_PATH
+        value: hosted-cellxgene-dev
+      - name: AWS_REGION
+        value: us-west-2
+      - name: AWS_DEFAULT_REGION
+        value: us-west-2
+      - name: API_DOMAIN
+        value: chief-puma.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
+      - name: WEB_DOMAIN
+        value: chief-puma.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
```
Comment (hspitzley-czi) [inline .infra/common.yaml:47]:
```suggestion
        value: lucky-bee.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
      - name: WEB_DOMAIN
        value: lucky-bee.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
```

File: .infra/common.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      - name: CXG_BUCKET_PATH
+        value: hosted-cellxgene-dev
+      - name: AWS_REGION
+        value: us-west-2
+      - name: AWS_DEFAULT_REGION
+        value: us-west-2
+      - name: API_DOMAIN
+        value: lucky-bee.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
+      - name: WEB_DOMAIN
+        value: lucky-bee.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
```
Comment (hspitzley-czi) [inline .infra/common.yaml:47]:
```suggestion
        value: $ARGOCD_APP_NAME.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
      - name: WEB_DOMAIN
        value: $ARGOCD_APP_NAME.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
```

File: .infra/common.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      - name: CXG_BUCKET_PATH
+        value: hosted-cellxgene-dev
+      - name: AWS_REGION
+        value: us-west-2
+      - name: AWS_DEFAULT_REGION
+        value: us-west-2
+      - name: API_DOMAIN
+        value: $ARGOCD_APP_NAME.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
+      - name: WEB_DOMAIN
+        value: $ARGOCD_APP_NAME.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
```
Comment (hspitzley-czi) [inline .infra/common.yaml:47]:
```suggestion
      - name: WEB_DOMAIN
        value: $ARGOCD_APP_NAME.dev-sc.dev.czi.team # TODO: be able to swap out stack name?
      - name: ARGOCD_APP_NAME
        value: $ARGOCD_APP_NAME
```

File: .infra/common.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      - name: CXG_BUCKET_PATH
+        value: hosted-cellxgene-dev
+      - name: AWS_REGION
+        value: us-west-2
+      - name: AWS_DEFAULT_REGION
+        value: us-west-2
+      - name: API_DOMAIN
+        value: $(__ARGUS_STACK_BASE_URL)
+      - name: WEB_DOMAIN
+        value: $(__ARGUS_STACK_BASE_URL)
```
Comment (hspitzley-czi) [inline .infra/common.yaml:47]:
```suggestion
        value: $(__ARGUS_STACK_INGRESS_HOST)
      - name: WEB_DOMAIN
        value: $(__ARGUS_STACK_INGRESS_HOST)
```

File: .github/workflows/argus-stack-prod-upsert.yaml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  prod_stack_upsert:
+    runs-on: [ARM64, self-hosted, Linux]
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Upsert Prod Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.17.1
```
Comment (hspitzley-czi) [inline .github/workflows/argus-stack-prod-upsert.yaml:16]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.19.1
```

File: .github/workflows/argus-stack-rdev-create.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  create-rdev-stack:
+    runs-on: [ARM64, self-hosted, Linux]
+    if: contains(github.event.pull_request.labels.*.name, 'stack')
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Create Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.17.1
```
Comment (hspitzley-czi) [inline .github/workflows/argus-stack-rdev-create.yaml:17]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.19.1
```

File: .github/workflows/argus-stack-rdev-delete.yaml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  delete-rdev-stack:
+    runs-on: [ARM64, self-hosted, Linux]
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'stack') || github.event.pull_request.state == 'closed' }}
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Delete Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/delete-stack@v0.17.1
```
Comment (hspitzley-czi) [inline .github/workflows/argus-stack-rdev-delete.yaml:17]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/delete-stack@v0.19.1
```

File: .github/workflows/argus-stack-staging-upsert.yaml [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  staging_stack_upsert:
+    needs: run_if
+    runs-on: [ARM64, self-hosted, Linux]
+    permissions:
+        id-token: write
+        contents: read
+
+    steps:
+      - name: Upsert Staging Stack
+        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.17.1
```
Comment (hspitzley-czi) [inline .github/workflows/argus-stack-staging-upsert.yaml:25]:
```suggestion
        uses: chanzuckerberg/argus-artifacts/ci/packages/create-stack@v0.19.1
```

**Review by tihuan — APPROVED** [review]

**Pull request merged**